erdjs/src/transaction.spec.ts

```diff
@@ -3,15 +3,17 @@ import { assert } from "chai";
 import { Transaction } from "./transaction";
 import * as errors from "./errors";
 import { Nonce } from "./nonce";
-import { GasLimit, GasPrice } from "./networkParams";
-
+import { ChainID, GasLimit, GasPrice, TransactionVersion } from "./networkParams";
+import { TransactionPayload } from "./transactionPayload";
+import { Balance } from "./balance";
+import { TestWallets } from "./testutils";
 
 describe("test transaction", () => {
-    let nonce: any = 42;
-    let gasLimit: any = 42;
-    let gasPrice: any = 42;
-
     it("should throw error when bad types", () => {
+        let nonce: any = 42;
+        let gasLimit: any = 42;
+        let gasPrice: any = 42;
+
         assert.throw(() => new Transaction({ nonce: nonce }), errors.ErrBadType);
```

Comment:
`any` is not allowed?

Reply:
Yup. We want to force the developer to use `new Nonce(42)` instead. In the constructor of `Transaction` we have the following checks:

```
guardType("nonce", Nonce, this.nonce);
guardType("gasLimit", GasLimit, this.gasLimit);
guardType("gasPrice", GasPrice, this.gasPrice);
```